File: BT1.py
from collections import deque

def function():
    m, n = map(int, input().split())
    ls = list(map(int, input().split()))
    ls += [0]
    stack = deque([(ls[0], 0)])
    result = 0
    for index, i in enumerate(ls[1:], start=1):
        k = index
        while stack and stack[-1][0] >= i:
            val, pos = stack.pop()
            if index - pos >= val:
                result = max(result, val * (index - pos))
            k = pos
        stack.append((i, k))

    for i in range(n):
        ls[i] = m - ls[i]
    ls[0] = 0
    stack = deque([(ls[0], 0)])
    for index, i in enumerate(ls[1:], start=1):
        k = index
        while stack and stack[-1][0] >= i:
            val, pos = stack.pop()
            if index - pos >= val:
                result = max(result, val * (index - pos))
            k = pos
        stack.append((i, k))    
    print(result)


if __name__ == "__main__":
    function()

File: BT1.cpp
#include <iostream>
#include <stack>
using namespace std;
int main(){   
    int t; cin >> t; 
    while (t--) {
        string s; cin >> s;
        stack<int> st;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '(') {
                st.push(i);
            }
            else if (s[i] == ')'){
                if(!st.empty()) {
                    s[st.top()] = '0';
                    s[i] = '1';
                    st.pop();
                }
            }

        }
         for(int i = 0 ; i< s.size(); i ++) {
        if (s[i] == '(' or s[i] == ')') cout << "-1";
        else cout << s[i];
    }
    cout << endl;
    }
   
}